### PR TITLE
Collect memory usage in transactions

### PR DESCRIPTION
**Pull request by stefanosiano**
## :scroll: Description
Added a MemoryCollector, with different implementations for Java and Android, set in SentryOptions
Added TransactionPerformanceCollector, that uses the MemoryCollector, to collect performance data while a transaction is running.
Collected data is a list of objects containing a timestamp, java heap memory usage and native memory usage (only on Android)

## :bulb: Motivation and Context
We want to build a POC to get the memory usage data as a time series while a transaction runs.

## :green_heart: How did you test it?
Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/AndroidMemoryCollector.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import android.os.Debug;
+import io.sentry.IMemoryCollector;
+import io.sentry.MemoryCollectionData;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public class AndroidMemoryCollector implements IMemoryCollector {
+  @Override
+  public MemoryCollectionData collect() {
+    long now = System.currentTimeMillis();
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/AndroidMemoryCollector.java:12]: Is `System.currentTimeMillis()` good enough as timestamp? Or should I go with `DateUtils.getCurrentDateTime().getTime()`?

File: sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          },
+          0,
+          TRANSACTION_COLLECTION_INTERVAL_MILLIS);
+    }
+  }
+
+  public @Unmodifiable List<MemoryCollectionData> stop(final @NotNull ITransaction transaction) {
+    synchronized (memoryMap) {
+      List<MemoryCollectionData> memoryData = memoryMap.remove(transaction.getEventId().toString());
+      if (memoryMap.isEmpty() && isStarted.get()) {
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java:76]: `m`: You can utilize `compareAndSet()` to read and update the AtomicBoolean at the same time
```suggestion
      if (memoryMap.isEmpty() && isStarted.compareAndSet(true, false)) {
```

File: sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        timer.cancel();
+        timer = new Timer();
+        isStarted.set(false);
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java:79]:
```suggestion
```

**Review by markushi — APPROVED** [review]
Nice one, looks good! Please have a look at my comments

**Pull request merged**